### PR TITLE
Added new DNS Scanner: Doggo

### PR DESCRIPTION
**Pull request by rseedorff**
## Description

Added a new DNS Scanner: Doggo 

### Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure that all your commits are signed-off and that you are added to the [Contributors](https://github.com/secureCodeBox/secureCodeBox/blob/main/CONTRIBUTORS.md) file.
* [x] Make sure that all CI finish successfully.
* [ ] Optional (but appreciated): Make sure that all commits are [Verified](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).

### CODE REVIEW

File: scanners/doggo/examples/example.com/scan.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+apiVersion: "execution.securecodebox.io/v1"
+kind: Scan
+metadata:
+  name: "doggo-localhost"
+spec:
+  scanType: "doggo"
+  parameters:
+    - "-invalidFlag"
```
Comment (J12934) [inline scanners/doggo/examples/example.com/scan.yaml:12]: examples should be working and not include invalid flags :)

File: scanners/doggo/parser/parser.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  return result;
+}
+
+function transformToFindings(targets) {
+  // Code to transform the scanner results to scb findings
+  return targets.map((target) => {
+    return {
+      name: `DNS Zone: ${target.name} | Type: ${target.type}`,
+      description: `DNS record type "${target.type}" found for zone "${target.name}" with address "${target.address}".`,
```
Comment (J12934) [inline scanners/doggo/parser/parser.js:28]: Is "zone" here really the correct / most helpful term?
Aren't these DNS records in a zone?

**Review by J12934 — COMMENTED** [review]
Look good 👍

I've added some missing some example findings and refactored the parser a bit to be more in line with our other parsers.
I got one question in the comments below which would be nice to clear up before merging

File: scanners/doggo/parser/parser.js (hunk 2 above)
Comment (rseedorff) [reply scanners/doggo/parser/parser.js]: Thx for the hint. Now fixed as discussed 👍🏻

**Review by Weltraumschaf — APPROVED** [review]

**Pull request merged**